l2geth/core/tx_pool.go

```diff
@@ -555,8 +555,11 @@ func (pool *TxPool) validateTx(tx *types.Transaction, local bool) error {
 	}
 	// Ensure the transaction adheres to nonce ordering
 	if rcfg.UsingOVM {
-		if pool.currentState.GetNonce(from) != tx.Nonce() {
+		nonce := pool.currentState.GetNonce(from)
+		if nonce > tx.Nonce() {
 			return ErrNonceTooLow
+		} else if nonce < tx.Nonce() {
+			return errors.New("nonce too high")
```

Comment:
we can use the concrete `ErrNonceTooHigh` error: https://github.com/ethereum-optimism/optimism/blob/8d67991aba584c1703692ea46273ea8a1ef45f56/l2geth/core/error.go#L34